src/main/java/com/sms/invoicify/service/CompanyService.java

```diff
@@ -1,23 +1,24 @@
 package com.sms.invoicify.service;
 
 import com.sms.invoicify.exception.InvoicifyCompanyExistsException;
-import com.sms.invoicify.exception.InvoicifyCompanyNotExistsException;
 import com.sms.invoicify.models.Address;
 import com.sms.invoicify.models.Company;
 import com.sms.invoicify.models.CompanyEntity;
-import com.sms.invoicify.models.CompanySummaryVO;
+import com.sms.invoicify.models.CompanySummary;
 import com.sms.invoicify.repository.CompanyRepository;
 import lombok.AllArgsConstructor;
 import org.springframework.stereotype.Service;
 
+import javax.transaction.Transactional;
 import java.util.List;
 import java.util.Optional;
 import java.util.stream.Collectors;
 
 @Service
+@Transactional
 @AllArgsConstructor
 public class CompanyService {
-  private final CompanyRepository companyRepository;
+  private CompanyRepository companyRepository;
```

Comment:
Are we sure we don't want to declare the repository as final?

Reply:
![image](https://user-images.githubusercontent.com/31148246/119416462-a9e50280-bcc1-11eb-9e64-3ab2a014c888.png)

See spring docs

https://docs.spring.io/spring-data/jpa/docs/1.5.0.RELEASE/reference/html/repositories.html